rclpy/rclpy/executors.py

```diff
@@ -544,14 +564,15 @@ def _wait_for_ready_callbacks(
             # Process ready entities one node at a time
             for node in nodes_to_use:
                 for tmr in node.timers:
-                    if tmr.timer_pointer in timers_ready:
-                        # Check that a timer is ready to workaround rcl issue with cancelled timers
-                        if _rclpy.rclpy_is_timer_ready(tmr.timer_handle):
-                            if tmr.callback_group.can_execute(tmr):
-                                handler = self._make_handler(
-                                    tmr, node, self._take_timer, self._execute_timer)
-                                yielded_work = True
-                                yield handler, tmr, node
+                    if tmr.handle.pointer in timers_ready:
+                        with tmr.handle as capsule:
```

Comment:
It almost seems like this line and the one above it should be swapped.  That is, we effectively want to "protect" the timer as a capsule, and *then* dereference it to get the pointer to ensure that the handle hasn't gone away in the meantime.  On the  other hand, this may not matter.  Thoughts?

Reply:
There's no need to protect a call to `handle.pointer`. It is returning a cached integer. Even if it was calling `PyCapsule_GetPointer` it would still be safe to call because the address held by a pycapsule is immutable.